app/models/course_user_datum.rb

```diff
@@ -228,6 +241,14 @@ def self.find_or_create_cud_for_course(course, uid)
       end
     end
   end
+  
+  # global grace left cache key
+  def ggl_cache_key
+    # gets it into the YYYYMMDDHHMMSS form
+    dua = course.cgdub_dependencies_updated_at.utc.to_s(:number)
+
+    "ggl/dua-#{dua}/u-#{id}"
```

Comment:
I know there's many places in the rest of the codebase that doesn't follow this, but in instance methods using self.<attribute> i.e self.id is more explicit than id, which also helps to prevent shadowing problems

Reply:
fixed! Thanks for the suggestion!